File: java/src/main/java/trinsic/security/ISecurityProvider.java
package trinsic.security;

import com.google.protobuf.InvalidProtocolBufferException;
import com.google.protobuf.Message;
import trinsic.okapi.DidException;
import trinsic.services.account.v1.AccountProfile;

public interface ISecurityProvider {
  String GetAuthHeader(AccountProfile accountProfile, Message message)
      throws InvalidProtocolBufferException, DidException;
}

File: dotnet/Trinsic/Security/ISecurityProvider.cs
using System.Threading.Tasks;
using Google.Protobuf;
using Trinsic.Services.Account.V1;

namespace Trinsic;

public interface ISecurityProvider
{
    Task<string> GetAuthHeaderAsync(AccountProfile accountProfile, IMessage message);

    string GetAuthHeader(AccountProfile accountProfile, IMessage message);
}